src/servers/sapphire_zone/Actor/Player.cpp

```diff
@@ -1786,3 +1751,33 @@ bool Core::Entity::Player::isOnEnterEventDone() const
 {
   return m_onEnterEventDone;
 }
+
+void Core::Entity::Player::setLandPermissions( uint8_t permissionSet, uint32_t permissionMask, int16_t landSetId, int16_t wardNum, int16_t zoneId )
+{
+  m_housePermission[permissionSet].landSetId = landSetId;
+  m_housePermission[permissionSet].permissionMask = permissionMask;
+  m_housePermission[permissionSet].wardNum = wardNum;
+  m_housePermission[permissionSet].worldId = 67;
+  m_housePermission[permissionSet].unkown1 = 0;
+}
+
+void Core::Entity::Player::sendLandPermissions()
+{
+  auto landPermissions = makeZonePacket< FFXIVIpcLandPermission >( getId() );
+
+  landPermissions->data().freeCompanyHouse = m_housePermission[Common::LandPermissionSlot::FreeCompany];
```

Comment:
I do not think permissions should generally be stored in the player.
Those are mostly implicit and should be stored in the land.